src/shared.rs

```diff
@@ -19,6 +20,8 @@ pub enum HyprError {
     FromUtf8Error(std::string::FromUtf8Error),
     /// Dispatcher returned non `ok` value
     NotOkDispatch(String),
+    /// Failed to parse an integer
+    ParseError(<i32 as FromStr>::Err),
```

Comment:
Please instead create an error using the new Hyprland-rs error types that are more detailed, and can be used to provide more insight

Reply:
also, even if you did use this, you should've used https://doc.rust-lang.org/std/num/struct.ParseIntError.html